src/plugins/intel_npu/src/plugin/src/plugin.cpp

```diff
@@ -513,6 +517,82 @@ Plugin::Plugin()
             _supportedProperties.emplace_back(ov::PropertyName(property.first, std::get<1>(property.second)));
         }
     }
+
+    checkForCompilerFallback();
+}
+
+void Plugin::checkForCompilerFallback() {
+    auto setDefaultCompilerType = [&](const ov::intel_npu::CompilerType compilerType) {
+        std::stringstream strStream;
+        strStream << compilerType;
+        _globalConfig.update({{std::string(ov::intel_npu::compiler_type.name()), strStream.str()}});
+    };
+
+    // Check if CID fallback is required
+    if (_backends->isCompilerFallbackRequired()) {
+        _logger.info("Driver Version is too old to use MLIR Compiler. "
+                     "Driver Compiler will be set as default compiler type.");
+        setDefaultCompilerType(ov::intel_npu::CompilerType::DRIVER);
+        return;
+    }
+
+    // The driver versions are fetched from zero device
+    const auto device = _backends->getDevice();
+    if (device == nullptr) {
+        return;
+    }
+
+    const std::optional<Version> driverElfVersion = device->getLibraryELFVersion();
+    const std::optional<Version> driverMIVersion = device->getLibraryMIVersion();
+
+    if (!(driverElfVersion && driverMIVersion)) {
+        // We don't need to set the Driver Compiler as default here,
+        // since only compatible drivers without 1.6 graph extension can pass Driver Version check.
```

Comment:
```suggestion
        // since only compatible drivers with 1.6 graph extension can pass Driver Version check.
```
If I got this right.

Reply:
Changed this part :(